modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DeploymentDirectoryWatcher.java

```diff
@@ -80,28 +83,27 @@
    * Property used to change the deployment mode to deploy only the indicated applications with no redeployment.
    * mule -M-Dmule.deploy.applications=app1:app2:app3
    * This can also be done passing an additional command line option (deprecated) like:
-   *  mule -app app1:app2:app3 will restrict deployment only to those specified apps.
+   * mule -app app1:app2:app3 will restrict deployment only to those specified apps.
    */
   public static final String DEPLOYMENT_APPLICATION_PROPERTY = "mule.deploy.applications";
 
   protected static final int DEFAULT_CHANGES_CHECK_INTERVAL_MS = 5000;
 
   protected transient final Logger logger = LoggerFactory.getLogger(getClass());
-
-  private final ReentrantLock deploymentLock;
-  private final ArchiveDeployer<Domain> domainArchiveDeployer;
   protected final ArchiveDeployer<Application> applicationArchiveDeployer;
   protected final Supplier<SchedulerService> schedulerServiceSupplier;
+  private final ReentrantLock deploymentLock;
+  private final ArchiveDeployer<Domain> domainArchiveDeployer;
   private final ArtifactTimestampListener<Application> applicationTimestampListener;
   private final ArtifactTimestampListener<Domain> domainTimestampListener;
   private final ObservableList<Application> applications;
   private final ObservableList<Domain> domains;
   private final DomainBundleArchiveDeployer domainBundleDeployer;
   private final File appsDir;
   private final File domainsDir;
-  private ScheduledExecutorService artifactDirMonitorTimer;
-
+  private final List<ArtifactStoppedDeploymentListener> artifactStoppedDeploymentListeners;
```

Comment:
use the interface (`ArtifactStoppedListener`) in the var declaration, not the impl